File: wms-lite-user-web/src/features/master/item/item.ts
/**
 * 품목(Item) 및 품목 카테고리(ItemCategory) 도메인 관련 타입 정의
 * 백엔드 wms-lite-server (com.wms.wms_lite.domain.master.item) DTO 스펙에 대응
 */

// 단위 (Unit) Enum
export type UnitType = 'EA' | 'BOX' | 'PALLET' | 'KG' | 'L' | 'M' | 'SET';

// 품목 사용 상태 Enum
export type ItemStatus = 'ACTIVE' | 'INACTIVE';

/**
 * 품목 카테고리 응답 DTO
 */
export interface ItemCategoryResponse {
  id: number;
  code: string;
  name: string;
  description?: string;
  status?: ItemStatus;
}

/**
 * 품목 단건 상세 응답 DTO (ItemResponse)
 */
export interface ItemResponse {
  id: number;
  code: string;
  name: string;
  barcode?: string;
  specification?: string;
  description?: string;
  supplierName?: string;
  supplierId?: number;
  categoryName?: string;
  categoryId?: number;
  unit: UnitType;
  status: ItemStatus;
  safetyStockQuantity?: number | null;
  createdAt: string;
  updatedAt: string;
}

/**
 * 품목 요약 목록 응답 DTO (ItemSummaryResponse)
 */
export interface ItemSummaryResponse {
  id: number;
  code: string;
  name: string;
  barcode?: string;
  specification?: string;
  supplierName?: string;
  categoryName?: string;
  unit: UnitType;
  status?: ItemStatus;
  safetyStockQuantity?: number | null;
}

/**
 * 품목 검색 및 필터 요청 DTO (ItemSearchRequest)
 */
export interface ItemSearchRequest {
  keyword?: string;
  supplierId?: number;
  categoryId?: number;
  status?: ItemStatus;
  page?: number;
  size?: number;
}

/**
 * 품목 신규 등록 요청 DTO (ItemCreateRequest)
 */
export interface ItemCreateRequest {
  code: string;
  name: string;
  barcode?: string;
  supplierId?: number;
  categoryId?: number;
  unit: UnitType;
  specification?: string;
  description?: string;
  safetyStockQuantity?: number | null;
}

/**
 * 품목 정보 수정 요청 DTO (ItemUpdateRequest)
 */
export interface ItemUpdateRequest {
  name: string;
  barcode?: string;
  supplierId?: number;
  categoryId?: number;
  unit: UnitType;
  specification?: string;
  description?: string;
  safetyStockQuantity?: number | null;
}

/**
 * 품목 상태 변경 요청 DTO (ItemStatusChangeRequest)
 */
export interface ItemStatusChangeRequest {
  status: ItemStatus;
}

File: wms-lite/wms-lite/src/main/java/com/wms/wms_lite/domain/master/item/entity/Item.java
package com.wms.wms_lite.domain.master.item.entity;

import com.wms.wms_lite.domain.master.item.enums.ItemStatus;
import com.wms.wms_lite.domain.master.item.enums.UnitType;
import com.wms.wms_lite.domain.master.supplier.entity.Supplier;
import com.wms.wms_lite.global.entity.SoftDeleteEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "items")
public class Item extends SoftDeleteEntity {

    @Column(nullable = false, unique = true, length = 50)
    private String code;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(length = 100)
    private String barcode;

    @Column(length = 200)
    private String specification;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private UnitType unit = UnitType.EA;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private ItemStatus status = ItemStatus.ACTIVE;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "supplier_id")
    private Supplier supplier;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private ItemCategory category;

    @Column
    private Integer safetyStockQuantity;
}
